File: WorldFeed/Server/src/Services/Post/WorldFeed.Post.Domain/AggregatesModel/PostAggregate/TwitterEntities/ExtendedEntities/VideoInformationEntity.cs
namespace WorldFeed.Post.Domain.AggregatesModel.PostAggregate.TwitterEntities.ExtendedEntities
{
    using Newtonsoft.Json;

    using WorldFeed.Upload.Domain.AggregatesModel.PostAggregate.Entities.ExtendedEntities;

    public class VideoInformationEntity : IVideoInformationEntity
    {
        [JsonProperty("aspect_ratio")]
        public int[] AspectRatio { get; set; }

        [JsonProperty("duration_millis")]
        public int DurationInMilliseconds { get; set; }

        [JsonProperty("variants")]
        public IVideoEntityVariant[] Variants { get; set; }
    }
}

File: Sharebook/Client/src/app/shared/post-object/Entities/ExtendedEntities/VideoInformationEntity.ts
import {IVideoInformationEntity} from "../interfaces/ExtendedEntities/IVideoInformationEntity";
import {IVideoEntityVariant} from "../interfaces/ExtendedEntities/IVideoEntityVariant";

export class VideoInformationEntity implements IVideoInformationEntity {
  // [JsonProperty("aspect_ratio")]
  public aspectRatio: number[];

  // [JsonProperty("duration_millis")]
  public durationInMilliseconds: number;

  // [JsonProperty("variants")]
  public variants: IVideoEntityVariant[];
}
